addons/mail/static/src/core/web/message_actions_patch.js

```diff
@@ -1,22 +1,106 @@
+import { getNonEditableMentions, parseEmail } from "@mail/utils/common/format";
 import { messageActionsRegistry } from "@mail/core/common/message_actions";
 import { _t } from "@web/core/l10n/translation";
+import { renderToMarkup } from "@web/core/utils/render";
+import { rpc } from "@web/core/network/rpc";
+
+function prepareActionBody(message) {
+    const email_from = message.author?.email ? message.author.email : message.email_from;
+    const [name, email] = parseEmail(email_from);
+    const datetimeFormatted = _t("%(date)s at %(time)s", {
+        date: message.datetime.toFormat("ccc, MMM d, yyyy"),
+        time: message.datetime.toFormat("hh:mm a"),
+    });
+    return renderToMarkup("mail.Message.bodyInForward", {
+        body: getNonEditableMentions(message.body),
+        date: datetimeFormatted,
+        email,
+        message,
+        name: name || email,
+    });
+}
+
+function prepareAction(name, context, message, thread) {
+    const actionContext = {
+        ...context,
+        default_composition_mode: "comment",
+        default_model: thread.model,
+        default_res_ids: [thread.id],
+        default_subject: message.subject || message.default_subject,
+        default_subtype_xmlid: "mail.mt_comment",
+    };
+    return {
+        name: name,
+        type: "ir.actions.act_window",
+        res_model: "mail.compose.message",
+        view_mode: "form",
+        views: [[false, "form"]],
+        target: "new",
+        context: actionContext,
+    };
+}
```

Comment:
This function is not worth it: it adds an indirection and important shaped data hidden in this function.
There's enough differences between both actions that it's best to dump the POJOs in the `onClick` of each action.